### ISSUE TITLE
fix: basic information tests from memote

### ISSUE THREAD

**Issue by BenjaSanchez** [SysBioChalmers/yeast-GEM#137]
### Description of the issue:
From the latest [memote report](https://github.com/SysBioChalmers/yeast-GEM/files/2191823/index_SBO.zip), the following basic information tests are giving unexpected results:
- [x] Only 1 unique metabolite: discussed in https://github.com/SysBioChalmers/yeast-GEM/issues/106#issuecomment-391685490
- [x] 684 reactions appear as duplicated (due to repeated annotation ids)
- [x] Almost all rxns are detected as transport
- [x] NGAM is not detected (the test errors)

**I hereby confirm that I have:**
- [x] Tested my code with [all requirements](https://github.com/SysBioChalmers/yeast-GEM#required-software---user) for running the model
- [x] Checked that a similar issue does not exist already
- [x] If needed, asked first in the [Gitter chat room](https://gitter.im/SysBioChalmers/yeast-GEM) about the issue

**Comment by feiranl** [comment]
Since we add 187 SLIMEr for lipid production which are recognized as unbalanced rxns, but as they are pseudo reactions, it is difficult to balance them, is that okay for memote to read SBO terms of those reactions and recognize that they are pseudo reactions and skip the balance test for those reactions? @ChristianLieven

**Comment by BenjaSanchez** [comment]
@feiranl note that this issue is pointed out in #135. Also, @ChristianLieven is on vacation until next year, but perhaps @Midnighter can help in this

**Comment by Midnighter** [comment]
So, we are already (as one factor of multiple heuristics) identifying different types of boundary reactions and the biomass reactions. So if you are annotating with SBO terms we can probably add the relevant terms to the list of reactions not to test. Using the SBO term in this way should be absolutely general, though, i.e., it should be clear that it is a term for an unbalanced pseudo reaction rather than pseudo-reactions in general.

**Comment by BenjaSanchez** [comment]
@Midnighter we use [SBO:0000395](http://www.ebi.ac.uk/sbo/main/SBO:0000395) (Encapsulating process) for these reactions. From the definition it sounds like reactions from that category would often be unbalanced, but I'm not sure if always. Should we create an issue in memote for further discussion?

**Comment by Midnighter** [comment]
Let's move the discussion to memote so that others may chime in. We already use the child terms of SBO:0000631.

> often created solely to complete model structure

This part does not sound fitting. I think 631 more describes reactions pseudoreactions that are needed purely for modeling purposes whereas lipid production describes actual biology just that it is unclear.

**Comment by BenjaSanchez** [comment]
note that SLIME reactions actually do not describe lipid production, they are just a formulation to represent lipid **requirements**, by splitting already produced lipids. So I think they could fit that description. But yes, let's continue in memote.

**Comment by feiranl** [comment]
Soory for that misleading information.

**Comment by BenjaSanchez** [comment]
@feiranl no worries! I have now created https://github.com/opencobra/memote/issues/537 for further discussion

**Comment by BenjaSanchez** [comment]
Update: All these issues are solved in yeast 8.4.2, as it can be see in its [memote report](https://github.com/SysBioChalmers/yeast-GEM/files/5629887/memote_matlab.zip), with the exception of the first problem (unique metabolites), which persists due to the official model ids not being BiGG. However, it is [solved](https://github.com/SysBioChalmers/yeast-GEM/files/5629942/memote_python.zip) if the model is transformed to be BiGG-compliant with the python wrapper:

  ```python
  import ComplementaryScripts.io as io
  model = io.read_yeast_model(make_bigg_compliant=True)  # load with BiGG ids
  io.write_yeast_model(model)  # save
  ```
